src/pip/_internal/commands/install.py

```diff
@@ -440,6 +441,18 @@ def run(self, options, args):
             message = create_os_error_message(
                 error, show_traceback, options.use_user_site,
             )
+            if WINDOWS and len(str(error)) >= 270:
+                if error.errno == errno.ENOENT:
```

Comment:
```suggestion
            if WINDOWS and error.errno == errno.ENOENT and len(str(error)) >= 270:
```

(And unindent the block.)

Reply:
Actually, `OSError` has a `filename` attribute, so we should ust that instead.

https://docs.python.org/3/library/exceptions.html#OSError

```python
if WINDOWS and error.errno == errno.ENOENT and error.filename and len(error.filename) >= 270:
```

The additional `error.filename` check makes sure the attribute is not `None`.

Reply:
So, am I supposed to add a 4th check which confirms that `error.filename` is not `None`? Or is it the suggested replacement for the second check above that is `error.errno == errno.ENOENT`?

Reply:
Makes sense I will get the changes pushed asap.